biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -1122,37 +1161,30 @@ public void run() {
 		}
 
 		Framework systemBundle;
-
 		if (parms.services) {
 			trace("using META-INF/services");
 			// 3) framework = null, lookup in META-INF/services
 
 			ClassLoader loader = getClass().getClassLoader();
 
-			// 3) Lookup in META-INF/services
-			List<String> implementations = getMetaInfServices(loader, FrameworkFactory.class.getName());
-
-			if (implementations.isEmpty())
-				error("Found no fw implementation");
-			if (implementations.size() > 1)
-				error("Found more than one framework implementations: %s", implementations);
-
-			String implementation = implementations.get(0);
-
-			Class<?> clazz = loader.loadClass(implementation);
-			FrameworkFactory factory = (FrameworkFactory) newInstance(clazz);
-			trace("Framework factory %s", factory);
 			@SuppressWarnings({
 				"unchecked", "rawtypes"
 			})
 			Map<String, String> configuration = (Map) p;
-			systemBundle = factory.newFramework(configuration);
-			trace("framework instance %s", systemBundle);
+
+			systemBundle = createConnect(loader, configuration);
+			if (systemBundle == null) {
+				systemBundle = createClassic(systemBundle, loader, configuration);
+			}
```

Comment:
It seems wrong to always try connect first. Equinox and Felix will eventually support connect and thus will have the service for connect. So this code will then always assume connect mode rather than normal OSGi framework mode. I think using connect must be some option on `-runfw` or some other bndrun file instruction.

Reply:
Nope, it only uses connect when it can find a Module Connector. Why would you put a Module Connector on the -runpath when you do not intend to use it?

Reply:
We should then doc this to make sure no one is surprised that a thing on runpath changes the launcher behavior.